### PR TITLE
Add `ssl_verify` option to OPA plugin

### PR DESCRIPTION
**Pull request by outsinre**
### Summary

Add `ssl_verify ` parameter to the `opa` plugin.

### Reason

The `opa` plugin now supports secure connection. Related Kong EE PR: https://github.com/Kong/kong-ee/pull/3271

`ssl_verify ` parameter controls if the OPA certificate is verified.

### CODE REVIEW

File: app/_hub/kong-inc/opa/_index.md [hunk 1]
```diff
@@ -79,6 +79,12 @@ params:
       default: false
       description: |
         If set to true, the Kong Gateway Consumer object in use for the current request (if any) is included as input to OPA.
+    - name: https_verify
+      required: true
+      datatype: boolean
+      default: true
+      description: |
+        Whether the OPA certificate will be verified according to the CA certificates specified in [lua_ssl_trusted_certificate](https://docs.konghq.com/gateway/latest/reference/configuration).
```
Comment (Guaris) [inline app/_hub/kong-inc/opa/_index.md:87]:
```suggestion
       If set to true, the OPA certificate will be verified according to the CA certificates specified in [lua_ssl_trusted_certificate](https://docs.konghq.com/gateway/latest/reference/configuration).
```

**Review by Guaris — CHANGES_REQUESTED** [review]

**Comment by mheap** [comment]
@outsinre Could you rebase on `release/gateway-3.0`?

**Comment by outsinre** [comment]
> @outsinre Could you rebase on `release/gateway-3.0`?

Sure, will do

**Comment by Guaris** [comment]
This will need to be rebased

**Comment by Guaris** [comment]
Any update on this? Do you need any help?

**Comment by outsinre** [comment]
> Any update on this? Do you need any help?

@Guaris, thanks for the notice. Just rebased!

**Comment by Guaris** [comment]
Is this still a draft?

**Comment by Guaris** [comment]
We can't merge this into the Gateway 3.0 branch because it would cause conflicts in that branch. Could I ask you to reset the head of this branch to the tip of Gateway-3.0, and cherry-pick your commits onto that. Essentially undoing the step where you rebased over the main branch. Alternatively, you could close this PR, create a new branch from the Gateway 3.0 branch, and add your docs again. I'm happy to help you with this on Slack if you need it.

**Comment by Guaris** [comment]
I fixed all of the branch issues for you :)

**Comment by mayocream** [comment]
@outsinre please rebase and clean up this PR.

**Review by mayocream — APPROVED** [review]

**Comment by mayocream** [comment]
@Guaris Could you help on merging this one? Thanks!

**Review by mayocream — APPROVED** [review]

**Comment by outsinre** [comment]
The Kong PR is already merged. Waiting for release.

**Pull request merged**